packages/headless/src/controllers/product-listing/sort/headless-product-listing-sort.ts

```diff
@@ -0,0 +1,66 @@
+import {SortCriterion} from '../../../features/sort-criteria/criteria';
+import {
+  buildCoreSort,
+  Sort,
+  SortInitialState,
+  SortProps,
+  SortState,
+} from '../../core/sort/headless-core-sort';
+import {fetchProductListing} from '../../../features/product-listing/product-listing-actions';
+import {ProductListingEngine} from '../../../app/product-listing-engine/product-listing-engine';
+import {parseCriterionExpression} from '../../../features/sort-criteria/criteria-parser';
+
+export interface ProductListingSortState extends SortState {
+  /**
+   * The current sort criteria, in a structured way.
+   */
+  sort: SortCriterion | SortCriterion[];
```

Comment:
Could we name the key `sortCriteria` for consistency?

Also, curious if the structured form is needed by UI devs? Could a string be enough, removing the need to extend interfaces?

Reply:
Additionally, I would prefer if we kept only one of the two props in this interface.

Reply:
We couldn't name it `sortCriteria` because it would clash with the `sortCriteria` string that's already in the state. However, we don't need to expose the structured form of `sort` for now.

I'm not too familiar with headless but could we create a selector on the parsed criterion expression (`parseCriterionExpression(getState().sortCriteria!)`) and then use the selector's result to make an api request to the product listings api? Would that be a possible better solution?

Reply:
> However, we don't need to expose the structured form of sort for now.
+1

> could we create a selector ...

Yep, a selector is just a function that takes in some state and returns a derived value. The `parseCriterionExpression` can be thought of as a selector,